thorlcr/activities/lookupjoin/thlookupjoinslave.cpp

```diff
@@ -2183,7 +2183,10 @@ class CHTBase
     }
     void setup(CSlaveActivity *activity, rowidx_t size, IHash *_leftHash, IHash *_rightHash, ICompare *_compareLeftRight)
     {
-        size32_t sz = sizeof(const void *)*size;
+        unsigned __int64 _sz = sizeof(const void *)*size;
+        if (_sz > UINT_MAX) // treat as OOM exception for handling purposes.
+            throw MakeStringException(ROXIEMM_MEMORY_LIMIT_EXCEEDED, "Unsigned overflow, trying to allocate hash table of size: %" I64F "d ", _sz);
+        size32_t sz = _sz;
```

Comment:
The memory manager supports > 2^32 byte allocations, so you should really be using a memsize_t here.  The check should be that the _sz = the cast value - which I think will almost always be true.
Also, should this be targetting 5.2?